### PR TITLE
Refactoring of configuration mechanism

### PR DESCRIPTION
**Pull request by dmytro-lazebnyi**

### CODE REVIEW

File: scalecube-organization/src/main/java/io/scalecube/config/ConfigRegistryConfiguration.java [hunk 1]
```diff
@@ -38,10 +30,13 @@ public static ConfigRegistry configRegistry() {
 
     // for test purposes without vault access
     if (System.getenv().get("VAULT_ADDR") != null) {
-      builder.addLastSource("vault", VaultConfigSource.builder().build());
+      builder.addFirstSource("vault", VaultConfigSource.builder().build());
```
Comment (segabriel) [inline scalecube-organization/src/main/java/io/scalecube/config/ConfigRegistryConfiguration.java]: it's confusing here, let's place it above of other declarations and leave it as `addLastSource`

File: scalecube-organization/src/main/resources/app.config.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+public.key.provider=io.scalecube.tokens.Auth0PublicKeyProvider
+token.expiration=2678400000
```
Comment (ronenhamias) [inline scalecube-organization/src/main/resources/app.config.properties]: token expiration should be "never" by default

meaning max long value.

Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/resources/app.config.properties]: Current default value was declared previously. 
In this pull request, I've renamed config file and rounded value of token.expiration. 
Do you think it should be changed from `2678400000` (which means 31 days) to `never expires` by default?

See
https://github.com/scalecube/scalecube-organization-service/blob/master/scalecube-organization/src/main/resources/settings.properties

Btw, I just found that there is a bug in token expiration calculation.

See https://github.com/scalecube/scalecube-organization-service/pull/72

Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/resources/app.config.properties]: The current default value was declared previously by Eran (@scooter72). 
In this pull request, I've renamed config file and rounded value of token.expiration. 
Do you think it should be changed from `2678400000` (which means 31 days) to `never expires` by default?

See
https://github.com/scalecube/scalecube-organization-service/blob/master/scalecube-organization/src/main/resources/settings.properties

Btw, I just found that there is a bug in token expiration calculation.

See https://github.com/scalecube/scalecube-organization-service/pull/72

Comment (ronenhamias) [reply scalecube-organization/src/main/resources/app.config.properties]: yes its will be a real problem if the token will expire after 31 days :)

Comment (dmytro-lazebnyi) [reply scalecube-organization/src/main/resources/app.config.properties]: Fixed in #72 (commit https://github.com/scalecube/scalecube-organization-service/pull/72/commits/008992d5a290ae5750af3c1037fccb8c2d54e7fa)

**Pull request merged**